packages/e2e-tests/testUtils.ts

```diff
@@ -231,11 +254,35 @@ export async function saveScreenshot(name: string) {
 
 export async function editViteConfig(replacer: (str: string) => string) {
 	editFile('vite.config.js', replacer);
-	if (!isBuild) {
+	if (testMode === 'serve') {
 		await waitForServerRestartAndPageReload();
 	}
 }
 
+export async function waitForBuildWatchAndPageReload(timeout = 10000) {
+	const logs = e2eServer.logs.watch.out;
+	const startIdx = logs.length;
+	let timeleft = timeout;
+	const pollInterval = 50;
+	let completed = false;
+	while (timeleft > 0) {
+		await sleep(pollInterval);
+		if (logs.some((text, i) => i > startIdx && text.match(/\d+ modules transformed\./))) {
```

Comment:
build watch logs the number of transformed modules each time a watch change triggered it.